### PR TITLE
[ConvNeXT] Fix drop_path_rate

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

As pointed out by #16699, the drop path rate attribute of `ConvNextConfig` wasn't implemented correctly. 

This PR fixes that, for both the PyTorch and Tensorflow implementations.

### CODE REVIEW

File: src/transformers/models/convnext/modeling_convnext.py [hunk 1]
```diff
@@ -209,8 +209,9 @@ class ConvNextEncoder(nn.Module):
     def __init__(self, config):
         super().__init__()
         self.stages = nn.ModuleList()
-        drop_path_rates = [x.item() for x in torch.linspace(0, config.drop_path_rate, sum(config.depths))]
-        cur = 0
```
Comment (NielsRogge) [inline src/transformers/models/convnext/modeling_convnext.py:213]: I've removed this (unclear) cur variable, and instead implement it similar to Ross' implementation in timm.

**Review by sgugger — COMMENTED** [review]
Agree on principle, but it looks this change is breaking the TF version of the model.

**Comment by NielsRogge** [comment]
Pinging @Rocketknight1 here.

**Review by Rocketknight1 — APPROVED** [review]
TF code looks good to me now!

**Comment by NielsRogge** [comment]
@sgugger ok for you?

**Review by sgugger — APPROVED** [review]

**Pull request merged**